uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/manage-leave-requests.js

```diff
@@ -132,7 +130,10 @@ define([
         classConfirm: 'btn-' + map[action].btnClass,
         msg: map[action].msg,
         onConfirm: function () {
-          return leaveRequest[action]();
+          return leaveRequest[action]()
+            .then(function () {
+              vm.refresh();
```

Comment:
vm.refresh does not need to return a promise, it does not need to be returned either. `then` returns a promise, so the following chains are valid:

```js
return leaveRequest[action]()
return leaveRequest[action]().then()
return leaveRequest[action]().then().then() // etc
```

As per the result, the onConfirm function does not utilise results of passed promises. Please check this file `org.civicrm.reqangular/src/common/controllers/dialog/dialog-ctrl.js`.